frontend/src/container/OnboardingContainer/Steps/DataSource/DataSource.tsx

```diff
@@ -74,12 +89,49 @@ export default function DataSource(): JSX.Element {
 		}
 	}, [selectedModule, selectedDataSource]);
 
+	const handleRequestDataSourceSubmit = async (): Promise<void> => {
+		try {
+			setIsSubmittingRequestForDataSource(true);
+			const response = await logEvent('Onboarding V2: Data Source Requested', {
+				module: selectedModule?.id,
+				dataSource: requestedDataSourceName,
+			});
+
+			if (response.statusCode === 200) {
```

Comment:
can be moved to separate component to avoid code duplication

Reply:
We will have multiple forms in that cases and we would required to share the reference back and forth to handle it.